core/app/mailers/spree/carton_mailer.rb

```diff
@@ -2,11 +2,13 @@ module Spree
   class CartonMailer < BaseMailer
     # Send an email to customers to notify that an individual carton has been
     # shipped.
-    def shipped_email(carton_id, resend: false)
-      @carton = Spree::Carton.find(carton_id)
+    def shipped_email(order, carton, resend: false)
```

Comment:
We can't fully maintain compatibility, since we can't determine which order was supposed to be sent to, but we could guess that it was always the first which would be correct for most stores.